lib/matplotlib/patches.py

```diff
@@ -80,6 +88,7 @@ def __init__(self, *,
                     "the edgecolor or facecolor properties.")
             self.set_color(color)
         else:
+            self.set_hatchcolor(hatchcolor)
```

Comment:
I think why this wouldn't work is because:
these methods are getting called using the `_internal_update` method in the bar or any other figure.

While we can make sure that it is called in the correct sequence internally in the init function, but I think we cannot make sure that the user sets the edgecolor first and then hatchcolor afterwards.

For example,

```py
plt.bar(x + 0.5 * width, y_blue, width, label="Height of blue bars", hatch=r"\\", facecolor=(0, 0, 0, 0), edgecolor="yellow", hatchcolor="blue")
```

In the above example, the set_edgecolor function is called first and then set_hatchcolor is called afterwards.

<br/>
<br/>

```py
plt.bar(x + 0.5 * width, y_blue, width, label="Height of blue bars", hatch=r"\\", facecolor=(0, 0, 0, 0), hatchcolor="blue", edgecolor="yellow")
```
In the above example, the set_hatchcolor function is called first and then set_edgecolor is called afterwards.